pyogrio/_io.pyx

```diff
@@ -1385,6 +1392,35 @@ def ogr_write(str path, str layer, str driver, geometry, field_data, fields,
                 elif field_type == OFTReal:
                     OGR_F_SetFieldDouble(ogr_feature, field_idx, field_value)
 
+                elif field_type == OFTDate:
+                    datetime = field_value.astype(object)
+                    OGR_F_SetFieldDateTimeEx(
+                        ogr_feature,
+                        field_idx,
+                        datetime.year,
+                        datetime.month,
+                        datetime.day,
+                        0,
+                        0,
+                        0.0,
+                        0
+                    )
+
+                elif field_type == OFTDateTime:
+                    # TODO: what to do with tzinfo?
+                    datetime = field_value.astype("datetime64[ms]").astype(object)
+                    OGR_F_SetFieldDateTimeEx(
+                        ogr_feature,
+                        field_idx,
+                        datetime.year,
+                        datetime.month,
+                        datetime.day,
+                        datetime.hour,
+                        datetime.minute,
+                        float(f"{datetime.second}.{datetime.microsecond:06d}"),
```

Comment:
Can you do this with a calculation (something like datetime.second + datetime.microsecond / 10**6) instead of string formatting+parsing? (or is there a specific reason for the latter?)

Reply:
That's indeed cleaner...